FLIR/conservator/local_dataset.py

```diff
@@ -915,21 +915,48 @@ def clone(dataset, clone_path=None, verbose=True, max_retries=5, timeout=5):
 
         return LocalDataset(dataset._conservator, clone_path)
 
+    def validate_annotations(self, index_data):
+        valid = True
+        for frame in index_data["frames"]:
+            target_ids = [a["targetId"] for a in frame["annotations"]]
+
+            # check for duplicate target IDs
+            unique_target_ids = set(target_ids)
+            if len(unique_target_ids) < len(target_ids):
+                logger.error(
+                    "Duplicate targetIds in annotations of frame %s",
+                    frame["datasetFrameId"],
+                )
+                valid = False
+
+            # check for negative target IDs
+            if any(int(tgt) < 0 for tgt in target_ids):
```

Comment:
This should be caught by the schema, in the form of a regex, not sure we need to check for this explicitly, either.

Reply:
It _is_ in the schema, but the problem is that it's a very end-user-unfriendly message, which caused a user to wish for an understandable message. This was an error that was apparently caused by an earlier annotation editor problem, so
* probably not going to be the only affected dataset
* not the "fault" of the user running their own custom script to update a dataset, so they end up really mystified how the dataset could be bad